nmpolicy/internal/capture/resolved_test.go

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2021 NMPolicy Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *	  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package capture_test
+
+import (
+	"testing"
+
+	assert "github.com/stretchr/testify/require"
+
+	"github.com/nmstate/nmpolicy/nmpolicy/internal/capture"
+)
+
+func TestResolvedCapture(t *testing.T) {
+	t.Run("Capture", func(t *testing.T) {
+		testResolveCaptureEntryPathExpression(t)
+
+		testResolveCaptureEntryPathLexFailure(t)
+		testResolveCaptureEntryPathParseFailure(t)
+		testResolveCaptureEntryPathResolveFailure(t)
+	})
+}
+
+func testResolveCaptureEntryPathExpression(t *testing.T) {
+	t.Run("resolve a capture entry path", func(t *testing.T) {
+		resolvedCapture := capture.NewResolvedCapture(lexerStub{}, parserStub{}, resolverStub{}, nil)
```

Comment:
This test is very unclear. It would be much more readable if it would explicitly pass the stub their return values.

Reply:
This can be done at follow up code here is new test with existing stubs.

Reply:
Changed so the input is used to create the output.